### PR TITLE
Fix issues on HyperV platform

### PR DESCRIPTION
**Pull request by LiliDeng**
Shorten OS VHD path in HyperV platform.

### CODE REVIEW

File: TestControllers/TestController.psm1 [hunk 1]
```diff
@@ -726,6 +726,9 @@ Class TestController {
 				$executionCount += 1
 				Write-LogInfo "($executionCount/$($this.TotalCaseNum)) testing started: $($currentTestCase.testName)"
 				Write-LogInfo "SetupConfig: { $(ConvertFrom-SetupConfig -SetupConfig $currentTestCase.SetupConfig) }"
+				if ($vmData -and $this.TestPlatform -imatch "HyperV") {
```
Comment (squirrelsc) [inline TestControllers/TestController.psm1:729]: Add comments above to explain why need this logic.

File: TestProviders/HyperVProvider.psm1 [hunk 2]
```diff
@@ -157,6 +157,8 @@ Class HyperVProvider : TestProvider
 				}
 			}
 
+			$allVMData = Get-AllHyperVDeployementData -HyperVGroupNames $allVMData[0].HyperVGroupName -CurrentTestData $CurrentTestData
```
Comment (squirrelsc) [inline TestProviders/HyperVProvider.psm1]: It's better to split to two commits. One is for path, another one is for refresh IPs.

File: TestControllers/TestController.psm1 (hunk 1 above)
Comment (squirrelsc) [inline TestControllers/TestController.psm1]: How about refresh for all platforms? It's better to keep upper level logic consistent.

**Review by squirrelsc — CHANGES_REQUESTED** [review]

File: TestProviders/HyperVProvider.psm1 (hunk 2 above)
Comment (squirrelsc) [reply TestProviders/HyperVProvider.psm1]: They can be two commits, but in the same PR. It depends on you.

**Review by squirrelsc — APPROVED** [review]

File: TestControllers/TestController.psm1 (hunk 1 above)
Comment (LiliDeng) [reply TestControllers/TestController.psm1]: IP won't change in Azure platform.

File: TestProviders/HyperVProvider.psm1 (hunk 2 above)
Comment (LiliDeng) [reply TestProviders/HyperVProvider.psm1]: Sorry, I open another PR #1148

**Pull request merged**